src/kokkos/test/kokkos/OneToManyAssoc_t.cc

```diff
@@ -0,0 +1,324 @@
+
+  std::mt19937 eng;
+  std::default_random_engine generator (1234);
+  std::geometric_distribution<int> rdm(0.8);
+  constexpr uint32_t rdm_N = 16000;
+  Kokkos::View<int*,KokkosExecSpace> rdm_d("rdm_d",rdm_N);
+  auto rdm_h = Kokkos::create_mirror_view(rdm_d);
+  for(uint32_t i=0;i<rdm_N;++i){
+    rdm_h(i) = rdm(eng);
+    //printf("00 rdm(eng) = %04d\n",rdm_h(i));
+  }
+  Kokkos::deep_copy(rdm_d,rdm_h);
+
+  constexpr uint32_t N = 4000;
+
+  Kokkos::View<uint16_t**,KokkosExecSpace> v_d("v_d",N,4);
+  auto v_h = Kokkos::create_mirror_view(v_d);
+  Kokkos::View<Assoc,KokkosExecSpace> a_d("a_d");
+  auto a_h = Kokkos::create_mirror_view(a_d);
+  Kokkos::View<SmallAssoc,KokkosExecSpace> sa_d("sa_d");
+  auto sa_h = Kokkos::create_mirror_view(sa_d);
+  Kokkos::View<AtomicPairCounter,KokkosExecSpace> dc_d("dc_d");
+  auto dc_h = Kokkos::create_mirror_view(dc_d);
+
+
+  Kokkos::View<long long,KokkosExecSpace> ave_d("ave_d");
+  auto ave_h = Kokkos::create_mirror_view(ave_d);
+  Kokkos::View<int,KokkosExecSpace> imax_d("imax_d");
+  auto imax_h = Kokkos::create_mirror_view(imax_d);
+  Kokkos::View<uint32_t,KokkosExecSpace> n_d("n_d");
+  auto n_h = Kokkos::create_mirror_view(n_d);
+  Kokkos::View<uint32_t,KokkosExecSpace> z_d("z_d");
+  auto z_h = Kokkos::create_mirror_view(z_d);
+  Kokkos::View<uint32_t,KokkosExecSpace> nz_d("nz_d");
+  auto nz_h = Kokkos::create_mirror_view(nz_d);
+  // fill with "index" to element
+  Kokkos::parallel_for("init",Kokkos::RangePolicy<KokkosExecSpace>(KokkosExecSpace(),0,1),
+    KOKKOS_LAMBDA(const int& i){
+      ave_d() = 0;
+      imax_d() = 0;
+      n_d() = 0;
+      z_d() = 0;
+      nz_d() = 0;
+
+      uint32_t rdm_counter = 0;
+
+      for (auto i = 0U; i < 4U; ++i) {
+        auto j = 0U;
+        while (j < N && n_d() < MaxElem) {
+          if (z_d() == 11) {
+            n_d() += 1;
+            z_d() = 0;
+            nz_d() += 1;
+            continue;
+          }  // a bit of not assoc
+          auto x = rdm_d(rdm_counter);
+          //printf("00 i = %04d j = %04d rdm(eng) = %04d\n",i,j,x);
+          rdm_counter++;
+          if(rdm_counter>=rdm_N)
+            printf("ERROR ran out of random numbers\n");
+          auto k = N;
+          if(j + x + 1 < N)
+            k = j + x + 1;
+          if (i == 3 && z_d() == 3) {  // some triplets time to time
+            for (; j < k; ++j)
+              v_d(j,i) = MaxElem + 1;
+          } else {
+            ave_d() += x + 1;
+            if(x > imax_d())
+              imax_d() = x;
+            for (; j < k; ++j)
+              v_d(j,i) = n_d();
+            n_d() += 1;
+          }
+          z_d() += 1;
+        }
+        assert(n_d() <= MaxElem);
+        assert(j <= N);
+      }
+    });
+
+  Kokkos::deep_copy(KokkosExecSpace(),n_h,n_d);
+  Kokkos::deep_copy(KokkosExecSpace(),nz_h,nz_d);
+  Kokkos::deep_copy(KokkosExecSpace(),ave_h,ave_d);
+  Kokkos::deep_copy(KokkosExecSpace(),imax_h,imax_d);
```

Comment:
This is the work around. In the CUDA version these variables `n`, `nz`, `ave`, and `imax` were initialized serially on the host device, then copied to their device versions. Since I had the issue with `deep_copy` of a `int16**` View (that triggered your finding `-isystem`)  I decided to initialize the device side objects instead so I could avoid the `deep_copy`.

Reply:
Ok, thanks for the explanation.I think it's good as it is. Could you add a short comment here about that?